File: server/greeter_service.go
package main

import (
    "log"

    pb "main/genproto"
)

// handleSayHello processes the HelloRequest and returns a HelloResponse
func handleSayHello(helloRequest *pb.HelloRequest, headers *map[string]string) (*pb.HelloResponse, error) {
    log.Printf("Received: %v", helloRequest.GetName())

    helloResp := &pb.HelloResponse{
        Text: "Hello " + helloRequest.GetName(),
    }

    return helloResp, nil
}

// handleSayBye processes the ByeRequest and returns a ByeResponse
func handleSayBye(byeRequest *pb.ByeRequest, headers *map[string]string) (*pb.ByeResponse, error) {
    log.Printf("Received: %v", byeRequest.GetName())

    byeResp := &pb.ByeResponse{
        Text: "Bye " + byeRequest.GetName(),
    }

    return byeResp, nil
}

File: examples/greeter/js/server/greeter_service.js
const {HelloRequest, HelloResponse, ByeRequest, ByeResponse} = require('./genproto');

function sayHello(helloRequest, headers) {
    console.log(`Received: ${helloRequest.getName()}`);

    const helloResp = new HelloResponse();
    helloResp.setText(`Hello ${helloRequest.getName()}`);

    return helloResp;
}

function sayBye(byeRequest, headers) {
    console.log(`Received: ${byeRequest.getName()}`);

    const byeResp = new ByeResponse();
    byeResp.setText(`Bye ${byeRequest.getName()}`);

    return byeResp;
}

module.exports = {sayHello, sayBye};
